components/cdc/src/endpoint.rs

```diff
@@ -350,14 +349,103 @@ impl ResolvedRegionHeap {
     fn is_empty(&self) -> bool {
         self.heap.is_empty()
     }
+}
 
-    fn clear(&mut self) {
-        self.heap.clear();
-    }
+#[derive(Default, Debug)]
+pub(crate) struct Advance {
+    // multiplexing means one region can be subscribed multiple times in one `Conn`,
+    // in which case progresses are grouped by (ConnId, request_id).
+    pub(crate) multiplexing: HashMap<(ConnId, u64), ResolvedRegionHeap>,
+
+    // exclusive means one region can only be subscribed one time in one `Conn`,
+    // in which case progresses are grouped by ConnId.
+    pub(crate) exclusive: HashMap<ConnId, ResolvedRegionHeap>,
+
+    // To be compatible with old TiCDC client before v4.0.8.
+    // TODO(qupeng): we can deprecate support for too old TiCDC clients.
```

Comment:
Is there any plan to deprecate support too old TiCDC clients? 

What condition satisfied can we do this ?

Reply:
We can do it right now. In the latest TiCDC, upstream cluster version should be greater than 6.1.